### PR TITLE
Add tests for API container delete

### PR DESCRIPTION
**Pull request by runcom**
Signed-off-by: Antonio Murdaca me@runcom.ninja

### CODE REVIEW

File: integration-cli/docker_api_containers_test.go [hunk 1]
```diff
@@ -1040,3 +1041,125 @@ func (s *DockerSuite) TestContainerApiCopyContainerNotFound(c *check.C) {
...
+
+	if _, err := os.Stat(vol); err != nil {
+		c.Fatal(err)
+	}
+
+	status, _, err := sockRequest("DELETE", "/containers/"+id+"?v=1&force=1", nil)
+	c.Assert(status, check.Equals, http.StatusNoContent)
+	c.Assert(err, check.IsNil)
+
+	if _, err := os.Stat(vol); !os.IsNotExist(err) {
```
Comment (tiborvass) [inline integration-cli/docker_api_containers_test.go]: use Assert

File: integration-cli/docker_api_containers_test.go [hunk 2]
```diff
@@ -1040,3 +1041,125 @@ func (s *DockerSuite) TestContainerApiCopyContainerNotFound(c *check.C) {
...
+
+func (s *DockerSuite) TestContainerApiDeleteRemoveVolume(c *check.C) {
+	testRequires(c, SameHostDaemon)
+
+	runCmd := exec.Command(dockerBinary, "run", "-d", "-v", "/testvolume", "busybox", "top")
+	out, _, err := runCommandWithOutput(runCmd)
+	c.Assert(err, check.IsNil)
+
+	id := strings.TrimSpace(out)
+	if err := waitRun(id); err != nil {
```
Comment (tiborvass) [inline integration-cli/docker_api_containers_test.go]: assert ?

**Comment by tiborvass** [comment]
@runcom for trivial err checks that don't add more context to the error message, you can just use assert directly, shorter.

File: integration-cli/docker_api_containers_test.go (hunk 2 above)
Comment (runcom) [inline integration-cli/docker_api_containers_test.go]: gotcha

File: integration-cli/docker_api_containers_test.go (hunk 1 above)
Comment (runcom) [inline integration-cli/docker_api_containers_test.go]: mmm not sure about this btw

Comment (tiborvass) [inline integration-cli/docker_api_containers_test.go]: fair enough

**Comment by runcom** [comment]
@tiborvass updated as per your previous comment

**Comment by tiborvass** [comment]
LGTM even if I think you forgot one https://github.com/docker/docker/pull/12948/files#diff-ca97ecb464e792b1086fca2ee8af23f8R1142

**Comment by runcom** [comment]
@tiborvass overlooked doing it right now

File: integration-cli/docker_api_containers_test.go (hunk 1 above)
Comment (runcom) [inline integration-cli/docker_api_containers_test.go]: How will this become with assert?
```
_, err = os.Stat(vol)                                                      
c.Assert(os.IsNotExist, check.Equals, false)
```
Do you like it? also tried `FitsTypeOf` from gocheck but really ugly

File: integration-cli/docker_api_containers_test.go [hunk 3]
```diff
@@ -1040,3 +1041,113 @@ func (s *DockerSuite) TestContainerApiCopyContainerNotFound(c *check.C) {
...
+	out, _, err := runCommandWithOutput(runCmd)
+	c.Assert(err, check.IsNil)
+
+	id := strings.TrimSpace(out)
+	c.Assert(waitRun(id), check.IsNil)
+
+	vol, err := inspectFieldMap(id, "Volumes", "/testvolume")
+	c.Assert(err, check.IsNil)
+
+	if _, err := os.Stat(vol); err != nil {
```
Comment (tiborvass) [inline integration-cli/docker_api_containers_test.go]: @runcom i'm talking aobut this one

Comment (runcom) [inline integration-cli/docker_api_containers_test.go]: Just done

**Comment by runcom** [comment]
ping @tiborvass @estesp tests green now

**Comment by estesp** [comment]
Yup..just saw that.  LGTM

**Pull request merged**